### PR TITLE
Filtering on empty cells by clicking on table is adding incorrect filter type

### LINKED ISSUE #23796: Filtering on empty cells by clicking on table is adding incorrect filter type
**Describe the bug**
Clicking on an empty cell and filtering on this value adds an incorrect type of filter. This has a couple side effects, the biggest being that the Save button will not be visible, and refreshing the page will simply redirect you to the original question you started from because the filter is not included in the serialized url.

**To Reproduce**
1. Go to a question that has a column with null/empty values
2. Click on a null / empty cell
3. Click the = to filter on this value
4. See the filter that is created will say "[Column] is equal to". This is implying we're checking `= null`

**Expected behavior**
I expect to see a filter where "[Column] is empty"

**Screenshots**
chrome_tiJRPhJl9D

**Information about your Metabase Installation:**
```
{
  "browser-info": {
    "language": "en-US",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/102.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.14.1+1-Ubuntu-0ubuntu1.20.04",
    "java.vendor": "Ubuntu",
    "java.vendor.url": "https://ubuntu.com/",
    "java.version": "11.0.14.1",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.14.1+1-Ubuntu-0ubuntu1.20.04",
    "os.name": "Linux",
    "os.version": "5.10.16.3-microsoft-standard-WSL2",
    "user.language": "en",
    "user.timezone": "America/Halifax"
  },
  "metabase-info": {
    "databases": [
      "h2",
      "postgres"
    ],
    "hosting-env": "unknown",
    "application-database": "h2",
    "application-database-details": {
      "database": {
        "name": "H2",
        "version": "1.4.197 (2018-03-18)"
      },
      "jdbc-driver": {
        "name": "H2 JDBC Driver",
        "version": "1.4.197 (2018-03-18)"
      }
    },
    "run-mode": "dev",
    "version": {
      "date": "2022-07-06",
      "src_hash": "16572ac7e6eeb0fa5e2986f3c30b1092f351d425",
      "tag": "v1.43.1-SNAPSHOT",
      "branch": "fixing-flaky-cypress-tests",
      "hash": "03d2884"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

### LINKED ISSUE #24149: Filtering with an empty cell using operators '<' and '>' returns an error in the table(question)
**Describe the bug** 
A table(question) that has empty cells, when you click on an empty cell to make some operation such as `<` or `>` it returns an error and these actions doesn't make sense since it is not possible to say if `[COLUMN]` is `less than` or `greater than` to `empty`.
 
 **Logs** 
 console_error
 
 **To Reproduce** 
1. Open some question that has some `empty` cell (e.g. Sample Database > Orders --- the "Discount" column has empty cells)
2. Click on an `empty` cell;
3. Click on `<` or `>`;
4. Now it should display the error.
 
 **Expected behavior** 
 When an `empty` cell is clicked, the actions buttons `<` and `>` should be **disabled** or **not be displayed**.
 
 **Demo** 
[Click here to see](https://www.loom.com/share/eab0e63b9d1e4acabec8080d49553e32)
 
 **Information about your Metabase Installation:**
 
Your browser and the version: **Opera GX 102 on Windows 11**
Your operating system: **Windows 11 64-bits**
Your databases: **H2**
Metabase version: **1.44.0-RC1**
Metabase hosting environment: **Unknown**
Metabase internal database: **1.4.197 (2018-03-18)**

**Additional context** 
NA

### PR DESCRIPTION
**Pull request by gitstart**
Fixes #23796 
Also Fixes #24149 

###### Before submitting the PR, please make sure you do the following

- [x] If you're attempting to fix a translation issue, please submit your changes to our [POEditor project](https://poeditor.com/join/project/ynjQmwSsGh) instead of opening a PR.

### Tests

- [x] Run the frontend and Cypress end-to-end tests with `yarn lint && yarn test`)
- [x] If there are changes to the backend codebase, run the backend tests with `clojure -X:dev:test`
- [x] Sign the [Contributor License Agreement](https://docs.google.com/a/metabase.com/forms/d/1oV38o7b9ONFSwuzwmERRMi9SYrhYeOrkbmNaq9pOJ_E/viewform)
      (unless it's a tiny documentation change).

### Demo
https://www.loom.com/share/fc54d4d94a614a76adb3ff7e50119c90

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/24302)

### CODE REVIEW

File: frontend/src/metabase-lib/lib/queries/structured/Filter.ts [hunk 1]
```diff
@@ -198,6 +207,10 @@ export default class Filter extends MBQLClause {
     return dimension && dimension.field();
   }
 
+  setOperatorName(operatorName: string) {
+    return (this[0] = operatorName);
+  }
```
Comment (iethree) [inline frontend/src/metabase-lib/lib/queries/structured/Filter.ts:212]: We shouldn't bypass the logic of `setOperator`

File: frontend/src/metabase-lib/lib/queries/structured/Filter.ts [hunk 2]
```diff
@@ -71,6 +71,15 @@ export default class Filter extends MBQLClause {
       const segment = this.segment();
       return segment ? segment.displayName() : t`Unknown Segment`;
     } else if (this.isStandard()) {
+      if (this.arguments()[0] === null) {
+        if (this.operatorName() === "=") {
+          this.setOperatorName("is-null");
+        }
+        if (this.operatorName() === "!=") {
+          this.setOperatorName("not-null");
+        }
```
Comment (iethree) [inline frontend/src/metabase-lib/lib/queries/structured/Filter.ts:80]: while we're at it, we should also handle `is-empty` and `not-empty`

**Review by iethree — CHANGES_REQUESTED** [review]

**Comment by gitstart** [comment]
@iethree Thank you for the review. Requested changes have been implemented, kindly look into it again

**Comment by nemanjaglumac** [comment]
Already fixed.

**Pull request closed without merging**